File: app/src/main/java/co/tpcreative/supersafe/common/adapter/BaseHolder.java
package co.tpcreative.supersafe.common.adapter;
import android.view.View;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.ButterKnife;

public class BaseHolder<V> extends RecyclerView.ViewHolder {
    public BaseHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }
    public void bind(V data, int position){}
    public void event(){}
}

File: app/src/main/java/co/tpcreative/supersafe/common/adapter/BaseHolder.kt
package co.tpcreative.supersafe.common.adapter
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import butterknife.ButterKnife

open class BaseHolder<V>(itemView: View) : RecyclerView.ViewHolder(itemView) {
    open fun bind(data: V, position: Int) {}
    fun event() {}
    init {
        ButterKnife.bind(this, itemView)
    }
}